File: tailoring-online-services/store-management-service/src/main/java/com/store_management_service/dto/ThreeDModelDto.java
package com.store_management_service.dto;

import lombok.*;

import java.io.Serializable;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ThreeDModelDto implements Serializable {
    private ProductDto productDto;
    private List<CustomizableMeasurementDto> measurementDtos;
    private List<CustomizableOptionDto> optionDtos;
}

File: tailoring-online-interfaces/src/app/core/dtos/three-d-model.dto.ts
import { CustomizableMeasurementDto } from "./customizable-measurement.dto";
import { CustomizableOptionDto } from "./customizable-option.dto";
import { ProductDto } from "./product.dto";

export interface ThreeDModelDto {
  productDto: ProductDto;
  measurementDtos: CustomizableMeasurementDto[];
  optionDtos: CustomizableOptionDto[];
}
